awx/main/models/notifications.py

```diff
@@ -434,32 +415,32 @@ def build_notification_message(self, nt, status):
         context = self.context(job_serialization)
 
         msg_template = body_template = None
+        msg = body = ''
 
+        # Use custom template if available
         if nt.messages:
             templates = nt.messages.get(self.STATUS_TO_TEMPLATE_TYPE[status], {}) or {}
-            msg_template = templates.get('message', {})
-            body_template = templates.get('body', {})
+            msg_template = templates.get('message', None)
+            body_template = templates.get('body', None)
+        # If custom template not provided, look up default template
```

Comment:
If no custom message is defined, look up the default template in the `notification_class` (see, for example, the [e-mail](https://github.com/ansible/awx/blob/devel/awx/main/notifications/email_backend.py#L23-L24) and [slack](https://github.com/ansible/awx/blob/devel/awx/main/notifications/slack_backend.py#L22) notification classes)